Zend/zend_builtin_functions.c

```diff
@@ -1500,7 +1500,12 @@ ZEND_FUNCTION(get_defined_constants)
 		} ZEND_HASH_FOREACH_END();
 		module_names[i] = "user";
 
-		ZEND_HASH_MAP_FOREACH_STR_KEY_PTR(EG(zend_constants), const_name, val) {
+		ZEND_HASH_MAP_FOREACH_PTR(EG(zend_constants), val) {
+			if (!val->name) {
```

Comment:
IIRC even special constants have a name, so this condition (and the one below) are a no-op

Reply:
Probably, but let's fix that in a separate PR as this is a clean revert.

Reply:
Sure! So at last you didn't have to fix anything?

Reply:
No, I think there must've been a bug in gcc that was patched.